midi_app_controller/state/state_manager.py

```diff
@@ -182,6 +186,7 @@ def select_controller(self, controller: Optional[SelectedItem]) -> None:
         Does not have any immediate effect except updating the value.
         """
         self.selected_controller = controller
+        self.select_recent_midi_ports()
```

Comment:
Users can access this method from their scripts etc. And they may not expect this method to automatically change their ports.
I think that it would be better if `select_recent_midi_ports` was called only in GUI (like `select_recent_binds`).